### PR TITLE
[INFRA] Add citation file and validation workflow

### PR DESCRIPTION
**Pull request by surchs**
Related to #147 

- added citation file
- added citation validation workflow

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# doc: https://citation-file-format.github.io/
+# schema: https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md
+
+cff-version: 1.2.0
+
+title: "neurobagel cli"
+
+version: 0.2.0
```
Comment (alyssadai) [inline CITATION.cff:8]:
```suggestion
version: 0.1.0
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# doc: https://citation-file-format.github.io/
+# schema: https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md
+
+cff-version: 1.2.0
+
+title: "neurobagel cli"
```
Comment (alyssadai) [inline CITATION.cff:6]: 🍒 How about something a little nicer ✨ ,
```suggestion
title: "Neurobagel CLI: Command line tool for Neurobagel data parsing and annotation"
```

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+version: 0.2.0
+
+abstract: "The Bagel CLI is a simple Python command line tool to automatically read and annotate a [BIDS dataset](https://bids-specification.readthedocs.io/en/stable/) so that it can be integrated into the Neurobagel graph."
```
Comment (alyssadai) [inline CITATION.cff:10]: hyperlinks don't seem to work properly in Zenodo descriptions - worth removing the BIDS spec URL?

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+message: "If you use this software, please cite it as below."
```
Comment (alyssadai) [inline CITATION.cff:12]: This appears as a little yellow box on the Zenodo page but I'm not sure if it's super necessary 🤷‍♀️

**Review by alyssadai — CHANGES_REQUESTED** [review]
Just a couple minor fixes!

File: CITATION.cff (hunk 1 above)
Comment (surchs) [reply CITATION.cff:8]: hehe, aha. daisy chainging mistakes is no good

File: CITATION.cff (hunk 4 above)
Comment (surchs) [reply CITATION.cff:12]: I'd leave it for now. Doesn't seem to hurt

File: CITATION.cff (hunk 3 above)
Comment (surchs) [reply CITATION.cff:10]: I think they want HTML style links, not markdown

**Review by alyssadai — APPROVED** [review]
👩‍🍳

File: CITATION.cff (hunk 3 above)
Comment (surchs) [reply CITATION.cff:10]: turns out they also don't like HTML links

**Pull request merged**